### ISSUE TITLE
xds: investigate io.envoyproxy.controlplane:api

### ISSUE THREAD

**Issue by dapengzhang0** [grpc/grpc-java#7117]
Similar to #6016 which was not possible because the latest version of `io.envoyproxy.protoc-gen-validate:pgv-java-stub` in maven is not up to date as what we needed.

`io.envoyproxy.controlplane:api` in maven is more up to date and even includes `com.google.cloudapi` that #7109 needs. However, we don't know how frequently it updates on a regular basis. It may still fall behind our need.

**Comment by ejona86** [comment]
`io.envoyproxy.controlplane:api` is completely broken, as it includes generated code for all the transitive dependencies. Those are published elsewhere. We shouldn't touch that with a ten-foot pole, and we should strongly discourage anyone else from using it.

**Comment by dapengzhang0** [comment]
I agree with you @ejona86 . My idea was still shading the envoy generated code and all its dependent generated code inside grpc-xds, not depending on io.envoyproxy.controlplane:api directly. The grpc-xds artifact will look the same as what we have now, just replacing the boilerplate of manually importing proto files.

**Comment by ejona86** [comment]
@dapengzhang0, that sounds fine. Just confirm that the protos are dropped (which I think we should be doing already).
